shell/platform/darwin/macos/framework/Source/FLEEngine.mm

```diff
@@ -345,6 +345,49 @@ - (void)sendOnChannel:(nonnull NSString*)channel message:(nullable NSData*)messa
   }
 }
 
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  FlutterPlatformMessageResponseHandle* response_handle = nullptr;
+  struct Captures {
+    FlutterBinaryReply reply;
+  };
+  auto captures = std::make_unique<Captures>();
+  captures->reply = callback;
+  auto message_reply = [](const uint8_t* data, size_t data_size, void* user_data) {
+    auto captures = reinterpret_cast<Captures*>(user_data);
+    NSData* reply_data = [NSData dataWithBytes:(void*)data length:data_size];
+    captures->reply(reply_data);
+    delete captures;
+  };
+
+  FlutterEngineResult result = FlutterPlatformMessageCreateResponseHandle(
+      _engine, message_reply, captures.get(), &response_handle);
+  if (result != kSuccess) {
+    NSLog(@"Failed to create a FlutterPlatformMessageResponseHandle");
+    return;
+  }
+  captures.release();
+
+  FlutterPlatformMessage platformMessage = {
+      .struct_size = sizeof(FlutterPlatformMessage),
+      .channel = [channel UTF8String],
+      .message = static_cast<const uint8_t*>(message.bytes),
+      .message_size = message.length,
+      .response_handle = response_handle,
+  };
+
+  result = FlutterEngineSendPlatformMessage(_engine, &platformMessage);
+  if (result != kSuccess) {
+    NSLog(@"Failed to send message to Flutter engine on channel '%@' (%d).", channel, result);
+  }
+
+  result = FlutterPlatformMessageReleaseResponseHandle(_engine, response_handle);
+    if (result != kSuccess) {
```

Comment:
Please run `clang-format` over this. I believe the presubmits will block this patch anyway because of the formatting. It is a good idea to `clang-format` on save of each engine file. You can setup your editor to do this so you don't forget.

Reply:
You fixed this right before I sent my review :) Thanks.

Reply:
:)